app/routes/my-actions.tsx

```diff
@@ -1,47 +1,52 @@
-import { NavLink } from "@remix-run/react";
+import type { LoaderFunction } from "@remix-run/node";
+import { NavLink, useLoaderData } from "@remix-run/react";
 import { Settings2 } from "lucide-react";
 
 import { cn } from "~/lib/utils";
 
+import History, { HistoryData } from "~/components/my-actions/history";
 import { SideBar } from "~/components/my-actions/sidebar";
 import { Summary } from "~/components/my-actions/summary";
 import { buttonVariants } from "~/components/ui/button";
 
-import History from "~/components/my-actions/history";
-import historyImg from "/sample-history-img.jpeg";
+import { getContributionsByAddress } from "~/directus.server";
+import { fetchReportByHCId } from "~/impact-reports.server";
 
-const mockData = [
-	{
-		id: 1,
-		date: new Date("2021-10-28"),
-		amount: 45.0,
-		img: {
-			src: historyImg,
-			alt: "sample history image",
-		},
-		title: "Construction of a Dobha in Giridih District",
-		category: "Dignity",
-		location: "Jammu and Kashmir",
-		description:
-			"lorem ipsum msg from the user after complete transaction, should just display max. 1 line here then editable on the report detail page.",
-	},
-	{
-		id: 2,
-		date: new Date("2021-06-18"),
-		amount: 145.0,
-		img: {
-			src: historyImg,
-			alt: "another sample history image",
-		},
-		title: "Construction of a well in Rishikesh",
-		category: "Thirst",
-		location: "Rishikesh",
-		description:
-			"lorem ipsum msg from the user after complete transaction, should just display max. 1 line here then editable on the report detail page.",
-	},
-];
+export const loader: LoaderFunction = async () => {
+	const history = [];
+	const testAddress = "0xf39Fd6e51aad88F6F4ce6aB8827279cfffb92266";
+
+	try {
+		const contributions = await getContributionsByAddress(testAddress);
+
+		for (const contribution of contributions) {
+			const report = await fetchReportByHCId(contribution.hypercert_id);
+
+			history.push({
+				id: contribution.txid,
+				date: new Date(contribution.date_created),
+				amount: contribution.amount,
+				img: {
+					src: report.image,
+					alt: report.title,
+				},
+				title: report.title,
+				category: report.category,
+				location: report.state,
+				description: report.summary,
+			} as HistoryData);
+		}
+	} catch (error) {
+		console.error(
+			`Failed to load reports contributed by ${testAddress}: ${error}`,
+		);
+	}
+	return history;
+};
```

Comment:
What happens if one of the reports fails during a loop? Would promise. Do you think promise.all makes sense here?

something like 
```ts
export const loader: LoaderFunction = async () => {
  const testAddress = "0xf39Fd6e51aad88F6F4ce6aB8827279cfffb92266";

  try {
    const contributions = await getContributionsByAddress(testAddress);

    // Map each contribution to a promise that resolves to the history item
    const historyPromises = contributions.map(async (contribution) => {
      const report = await fetchReportByHCId(contribution.hypercert_id);

      return {
        id: contribution.txid,
        date: new Date(contribution.date_created),
        amount: contribution.amount,
        img: {
          src: report.image,
          alt: report.title,
        },
        title: report.title,
        category: report.category,
        location: report.state,
        description: report.summary,
      };
    });

    // Wait for all promises to resolve and assign directly to history
    const history = await Promise.all(historyPromises);

    return history;
  } catch (error) {
    console.error(`Failed to load reports contributed by ${testAddress}: ${error}`);
    // Return an empty array or handle the error as needed
    return [];
  }
};
```

Reply:
reflected in 66dc3c7